Sources/SashaCore/Services/IconService.swift

```diff
@@ -240,6 +231,33 @@ final class IconService {
             try file.write(data: outputData)
         }
     }
+
+    /// Checks icon and image size compatibility and returns the scale factor between them
+    ///
+    /// - Parameters:
+    ///   - image: The original image.
+    ///   - icon: Object that represents icon information.
+    /// - Throws: `IconService.Error` errors.
+    func scaleIcon(_ icon: IconRepresentable, with image: CIImage) throws -> Float {
```

Comment:
`IconRepresentable` is a custom type, so we can omit its name in the function name
```suggestion
    func scale(_ icon: IconRepresentable, with image: CIImage) throws -> Float {
```